apps/sms/views/conversation/js/conversation.js

```diff
@@ -2538,26 +2544,26 @@ var ConversationView = {
 
       if (errorCode) {
         this.showMessageError(errorCode, {
-          confirmHandler: function stateResetAndRetry() {
-            var serviceId = Settings.getServiceIdByIccId(iccId);
-            if (serviceId === null) {
-              // TODO Bug 981077 should change this error message
-              this.showMessageError('NoSimCardError');
-              return;
-            }
-
+          confirmHandler: () => {
             // TODO move this before trying to call retrieveMMS in Bug 981077
             // Avoid user to click the download button while sim state is not
             // ready yet.
-            this.setMessageStatus(id, 'pending');
-            button.setAttribute('data-l10n-id', 'downloading-attachment');
-            Settings.switchMmsSimHandler(serviceId).then(
-              this.retrieveMMS.bind(this, messageDOM))
-            .catch(function(err) {
-                err && console.error(
-                  'Unexpected error while retrieving the MMS message', err);
+            setMessageButtonStatus('pending');
+            MozMobileConnsClient.switchMmsSimHandler(iccId).then(() =>
+              this.retrieveMMS.bind(this, messageDOM)
+            ).catch((err) => {
+              setMessageButtonStatus('error');
+
+              if (err === 'NoSimCardError') {
+                // TODO Bug 981077 should change this error message
+                this.showMessageError(err);
+                return;
+              }
+
+              err && console.error(
```

Comment:
__nit:__ mmm even if error object is not defined, we still likely have unexpected error, so maybe ```console.error('Unexpected error while retrieving the MMS message', err)```?